libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/MapboxTripSession.kt

```diff
@@ -182,12 +185,16 @@ internal class MapboxTripSession(
         if (state == TripSessionState.STOPPED) {
             return
         }
-        tripService.stopService()
-        stopLocationUpdates()
-        ioJobController.job.cancelChildren()
-        mainJobController.job.cancelChildren()
-        reset()
-        state = TripSessionState.STOPPED
+
+        stopSessionJob = threadController.getMainScopeAndRootJob().scope.launch {
```

Comment:
I'm wondering what other consequences can this have. Previously, stopping location updates, notifying session state listeners, canceling other coroutines, etc were all done synchronously, when `MapboxNavigation#stopTripSession` was called.

Now, we are waiting for a result of the `setRouteJob` before we do those operations. For example, if someone calls `stopTripSession` in `Activitiy#onStop`, we could actually end up delivering another location update to them _after_ `#onStop`, isn't that right? This could lead to invalid view updates, leaks, or other consequences.

Reply:
so how we can be sure that `setRoute` is finished?

Reply:
The setting the route doesn't actually have to be canceled or finished when we stop the trip session. It's not related to that lifecycle. What if we keep a separate scope for setting the route, that's not canceled when `stopTripSession` is called?

Reply:
I agree with Lukas that blocking usually has consequences. but i'm also unsure what exactly what the separate scope would be. @LukasPaczos can you provide more details on the idea?

When I was trying to figure out what you mean, had another idea and suggested it over here 
https://github.com/mapbox/mapbox-navigation-android/pull/3864/files#r545340723

Reply:
I was thinking about `setRoute` having a separate job like it's proposed in this PR, but we wouldn't cancel that job when the trip session is stopped since stopping the session is unrelated to setting the route.

Reply:
> I was thinking about setRoute having a separate job like it's proposed in this PR, but we wouldn't cancel that job when the trip session is stopped since stopping the session is unrelated to setting the route.

incorporated this suggestion into https://github.com/mapbox/mapbox-navigation-android/pull/3919 and made a non blocking solution. a separate job was needed to allow setRoute to be called even when the session has been stopped. the separate job is considered to be canceled by the [ThreadController.cancelAll](https://github.com/mapbox/mapbox-navigation-android/blob/dafada2006df4af83171761df79076a70899a016/libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt#L376-L377) calls.